vllm/engine/llm_engine.py

```diff
@@ -711,3 +717,42 @@ def _run_workers(
         for other_output in all_outputs[1:]:
             assert output == other_output
         return output
+
+    def _execute_model_dag(
+        self,
+        seq_group_metadata_list=None,
+        blocks_to_swap_in=None,
+        blocks_to_swap_out=None,
+        blocks_to_copy=None,
+        get_all_outputs: bool = False,
+    ) -> Any:
+        """Runs the given method on all workers using static DAG APIs."""
+        from ray.dag import OutputNode, InputNode
+        assert self.parallel_config.worker_use_ray
+
+        all_outputs = []
+        with InputNode() as input_data:
+            forward_dag = OutputNode([
+                worker.execute_method.bind(
+                    "execute_model",
+                    seq_group_metadata_list=input_data.seq_group_metadata_list,
```

Comment:
Note: It is currently not possible to pass *args, **kwargs to input dag. I haven't spent much time to enable it, but it wasn't trivial